### ISSUE TITLE
QGIS 3.4.2 - timemanager 3.0.2 - netcdf raster

### ISSUE THREAD

**Issue by pogam** [anitagraser/TimeManager#273]
I am unable to load netcdf raster in timemanager (v3.0.2) in QGIS v3.4.2
I am using the netcdf file from  [sample data](https://drive.google.com/file/d/0Bwc-5JFVTnfIb3E2LVR4NjE0SjA/view) showing cloud over spain.

following the discussion of [stack echange](https://gis.stackexchange.com/questions/306306/qgis-3-4-2-timemanager-plugin-netcdf-attribute?noredirect=1#comment495721_306306), this could be link with this [open issue](https://github.com/anitagraser/TimeManager/issues/256). 

below I reported the message from python and the timemanager plugin:

this the log message of the Timemanager

> 2018-12-18T17:30:29     INFO    Plugin language loaded: en
> 2018-12-18T17:30:29     CRITICAL    Translation file {} for lang {} was not loaded properly,falling back to English
> 2018-12-18T17:30:29     INFO    TimeManager: $Id$ loaded!
> 2018-12-18T17:31:13     WARNING    Could not import netcdftime. Using fallback computation
> 
> 

the is the python warning I received

> 2018-12-18T17:30:29     WARNING    warning:/usr/lib/python3/dist-packages/qgis/utils.py:672: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>              
>              traceback: File "", line 1, in 
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 335, in startPlugin
>               plugins[packageName] = package.classFactory(iface)
>               File "/home/paugam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/timemanager/__init__.py", line 15, in classFactory
>               from .timemanager import timemanager
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/home/paugam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/timemanager/timemanager.py", line 26, in 
>               from .timemanagercontrol import TimeManagerControl
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/home/paugam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/timemanager/timemanagercontrol.py", line 20, in 
>               from .timelayerfactory import TimeLayerFactory
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/home/paugam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/timemanager/timelayerfactory.py", line 15, in 
>               from .raster.wmstlayer import WMSTRasterLayer
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/home/paugam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/timemanager/raster/wmstlayer.py", line 1, in 
>               from future import standard_library
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 1023, in _handle_fromlist
>               File "", line 219, in _call_with_frames_removed
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/usr/lib/python3/dist-packages/future/standard_library/__init__.py", line 66, in 
>               import imp
>               File "/usr/lib/python3/dist-packages/qgis/utils.py", line 672, in _import
>               mod = _builtin_import(name, globals, locals, fromlist, level)
>               File "", line 971, in _find_and_load
>               File "", line 955, in _find_and_load_unlocked
>               File "", line 665, in _load_unlocked
>               File "", line 678, in exec_module
>               File "", line 219, in _call_with_frames_removed
>               File "/usr/lib/python3.6/imp.py", line 33, in 
>               DeprecationWarning, stacklevel=2)

**Comment by pogam** [comment]
while waiting for this bug to be fixed, do you know if I could used another raster format that could work in timemanager. 
thanks.

**Comment by pogam** [comment]
I have been looking at the option to load multiple tiff file, one for each of my time stamp. 
Is there a way to load raster layer in timemanager via a script

I tried something like the script below but it does not load the layer in the timemanager
Thanks for any advice
Ronan

```
import datetime

import timemanager
import timemanager.layer_settings as ls
from timemanager.timerasterlayer import TimeRasterLayer
from timemanager.time_util import DEFAULT_FORMAT, UTC, datetime_to_epoch

for layer in QgsProject.instance().mapLayers().values():
    print (layer.type)
    if layer.type() != QgsMapLayer.RasterLayer: continue
    
    print (layer.name())
    
    settings = ls.LayerSettings()
    settings.layer = layer
    settings.startTimeAttribute = "1970-01-01 00:00:01"
    settings.endTimeAttribute = "1970-11-01 06:45:26"
    raster = TimeRasterLayer(settings, iface=layer)

    assert (raster.getTimeFormat() == DEFAULT_FORMAT)

    raster.setTimeRestriction(datetime.datetime(1970, 1, 1, 0, 0, 2), datetime.timedelta(minutes=5))
    
print ('done'`)
```

**Comment by anitagraser** [comment]
I'm not sure if there is a good way to add layers from outside TimeManager. It would be more useful to spend the time on actually fixing this issue.

**Comment by pogam** [comment]
would you have a test example I could run to mimic the issue with the cloud.nc file present in the datatest directory?
I could give a try.

**Comment by anitagraser** [comment]
Please clarify, I'm not sure what you need. I don't have any nc files beyond the could cover in the sample directory.

**Comment by anitagraser** [comment]
#256 is fixed in the current master version. Please try and report back.

**Comment by pogam** [comment]
I found a bug that fix my issue loading netcdf
I worked on the master version with #256 fixed
the bug was in the string format of the band name 
i just did a pull request

**Comment by anitagraser** [comment]
Ok, it actually fails at netcdftime

**Comment by cgohlke** [comment]
> Ok, it actually fails at netcdftime and there are no wheels for it on https://www.lfd.uci.edu/~gohlke/pythonlibs/

netcdftime was replaced by cftime some time ago. Anyway, I put [netcdftime wheels](https://www.lfd.uci.edu/~gohlke/pythonlibs/#netcdftime) back for now.

**Comment by pogam** [comment]
the issue with netcdf4 is another problem
the fallback route is chosen instead when the netCDF4 is not installed 
on a ubuntu system `sudo apt-get install python3-netcdf4` install correctly netcdf4

the modification I added is independent of this
it ensures the re.findall to read the time from the bandname variable in raster/cdflayer.py

**Comment by carolinux** [comment]
fixed by #276 by @pogam

**Comment by carolinux** [comment]
fixed by #276 by @pogam
